dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/SubSnpCoreFields.java

```diff
@@ -46,18 +48,19 @@
      * @param chromosomeStart Start coordinate in chromosome, null if the contig is not fully mapped to a chromosome
      * @param chromosomeEnd End coordinate in chromosome, null if the contig is not fully mapped to a chromosome
      */
-    public SubSnpCoreFields(long ssId, Long rsId, int snpOrientation, String contig, int contigStart, int contigEnd,
-                            int contigOrientation, String chromosome, Integer chromosomeStart, Integer chromosomeEnd) {
+    public SubSnpCoreFields(long ssId, Integer rsId, int snpOrientation, String contig, int contigStart, int contigEnd,
+                            int contigOrientation, String chromosome, Long chromosomeStart, BigDecimal chromosomeEnd) {
         if (contigStart < 0 || contigEnd < 0) {
             throw new IllegalArgumentException("Contig coordinates must be non-negative numbers");
         }
-        if ((chromosomeStart != null && chromosomeStart < 0) || (chromosomeEnd != null && chromosomeEnd < 0)) {
+        if ((chromosomeStart != null && chromosomeStart < 0)
+                || (chromosomeEnd != null && chromosomeEnd.compareTo(new BigDecimal(0)) == 0)) {
```

Comment:
This is not the same condition, it should be `chromosomeEnd.compareTo(new BigDecimal(0)) < 0`